src/utils/DatabaseFilter.cpp

```diff
@@ -0,0 +1,167 @@
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Oliver Alka $
+// $Authors: Oliver Alka $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/FORMAT/FASTAFile.h>
+#include <OpenMS/METADATA/PeptideIdentification.h>
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+//Doxygen docu
+//-------------------------------------------------------------
+
+/**
+    @page UTILS_DatabaseFilter DatabaseFilter
+
+    @brief Tool to create a reduced database from a standard FASTA database.
+
+    A FASTA database serves as input and will be filtered depending on a give reverence vector.
+
+    Given entries are pasted to the new reduced database.
+
+    Filter whitelist: Input whitelist as reverence; Output database with entries on whitelist.
+
+    Filter blacklist: Input blacklist as reverence; Output database without entries on blacklist.
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude UTILS_DatabaseFilter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude UTILS_DatabaseFilter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+class TOPPDatabaseFilter :
+  public TOPPBase
+{
+public:
+  TOPPDatabaseFilter() :
+    TOPPBase("DatabaseFilter", "Tool to create a reduced database.", false)
+  {
+
+  }
+
+protected:
+  void registerOptionsAndFlags_()
+  {
+    registerInputFile_("in", "<file>", "","Input FASTA file, containing a database.");
+    setValidFormats_("in", ListUtils::create<String>("fasta"));
+    registerInputFile_("accession", "<file>", "", "Input IdXML file, containing the identfied peptides.", true);
+    setValidFormats_("accession", ListUtils::create<String>("idXML"));
+    registerStringOption_("method", "<type>", "whitelist", "Switch between white/blacklisting", false);
```

Comment:
please explain in description what this does. Often people will only look at the parameter description on CLI or in TOPPAS and therefore have trouble to understand what this parameter will do